contracts/AxelarGateway.sol

```diff
@@ -252,67 +235,103 @@ abstract contract AxelarGateway is IAxelarGateway, AdminMultisigBase {
     }
 
     /**********************\
-    |* Internal Functions *|
+    |* External Functions *|
     \**********************/
 
-    function _burnTokenFrom(
-        address sender,
-        string memory symbol,
-        uint256 amount
-    ) internal {
-        address tokenAddress = tokenAddresses(symbol);
+    function setup(bytes calldata params) external override {
+        // Prevent setup from being called on a non-proxy (the implementation).
+        if (implementation() == address(0)) revert NotProxy();
 
-        if (tokenAddress == address(0)) revert TokenDoesNotExist(symbol);
-        if (amount == 0) revert InvalidAmount();
+        (address[] memory adminAddresses, uint256 newAdminThreshold, bytes memory operatorshipParams) = abi.decode(
+            params,
+            (address[], uint256, bytes)
+        );
 
-        TokenType tokenType = _getTokenType(symbol);
-        bool burnSuccess;
+        // NOTE: Admin epoch is incremented to easily invalidate current admin-related state.
+        uint256 newAdminEpoch = _adminEpoch() + uint256(1);
+        _setAdminEpoch(newAdminEpoch);
+        _setAdmins(newAdminEpoch, adminAddresses, newAdminThreshold);
 
-        if (tokenType == TokenType.External) {
-            burnSuccess = _callERC20Token(
-                tokenAddress,
-                abi.encodeWithSelector(IERC20.transferFrom.selector, sender, address(this), amount)
-            );
+        if (operatorshipParams.length > 0) {
+            IAxelarAuthModule(AUTH_MODULE).transferOperatorship(operatorshipParams);
 
-            if (!burnSuccess) revert BurnFailed(symbol);
-
-            return;
+            emit OperatorshipTransferred(AUTH_MODULE, params);
```

Comment:
Should create an internal function to avoid duplicating these.

Reply:
Yup already done in my branch 👍